src/anndata/_io/specs/methods.py

```diff
@@ -629,11 +629,16 @@ def write_vlen_string_array_zarr(
 
         dataset_kwargs = dataset_kwargs.copy()
         dataset_kwargs = zarr_v3_compressor_compat(dataset_kwargs)
-        filters, dtype = (
-            ([VLenUTF8()], object)
-            if ad.settings.zarr_write_format == 2
-            else (None, str)
-        )
+        match (
+            ad.settings.zarr_write_format,
+            Version(np.__version__) >= Version("2.0.0"),
+        ):
+            case 2, _:
+                filters, dtype = [VLenUTF8()], object
+            case 3, True:
+                filters, dtype = None, np.dtypes.StringDType()
+            case 3, False:
+                filters, dtype = None, np.dtypes.ObjectDType()
```

Comment:
Not sure how to force coverage here aside from using a new test job altogether

Reply:
We could add a no-cover but that also feels wrong somehow, but maybe it's the best way forward.  It's not that the code is unreachable (which is usually what I've seen no-cover for) but rather that we don't test it

Reply:
and we can’t/shouldn’t/don’t test it because … ?

“it’s a bunch of work for a tiny benefit” is a totally acceptable answer.

Reply:
It's actually minimal work.  We would just need to special case a test job for this where we do "numpy minimum" + "zarr maximum"

Reply:
But do we want to add an entire CI job and/or infrastructure just for that? Not sure.

Reply:
got it. OK, no test it is!